Zend/Optimizer/zend_func_infos.h

```diff
@@ -38,12 +55,40 @@ static const func_info_t func_infos[] = {
     F1("stream_socket_accept", MAY_BE_RESOURCE|MAY_BE_FALSE),
     FN("zip_open", MAY_BE_RESOURCE|MAY_BE_LONG|MAY_BE_FALSE),
     FN("zip_read", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    F1("json_encode", MAY_BE_STRING|MAY_BE_FALSE),
+    F1("json_last_error_msg", MAY_BE_STRING),
     FN("oci_new_connect", MAY_BE_RESOURCE|MAY_BE_FALSE),
     FN("oci_connect", MAY_BE_RESOURCE|MAY_BE_FALSE),
     FN("oci_pconnect", MAY_BE_RESOURCE|MAY_BE_FALSE),
     FN("oci_parse", MAY_BE_RESOURCE|MAY_BE_FALSE),
     FN("oci_get_implicit_resultset", MAY_BE_RESOURCE|MAY_BE_FALSE),
     FN("oci_password_change", MAY_BE_RESOURCE|MAY_BE_BOOL),
     FN("oci_new_cursor", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_prepare", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_exec", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_connect", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_pconnect", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_tables", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_columns", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_gettypeinfo", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_primarykeys", MAY_BE_RESOURCE|MAY_BE_FALSE),
+#if !defined(HAVE_SOLID) && !defined(HAVE_SOLID_30) && !defined(HAVE_SOLID_35)
+    FN("odbc_procedurecolumns", MAY_BE_RESOURCE|MAY_BE_FALSE),
+#endif
+#if !defined(HAVE_SOLID) && !defined(HAVE_SOLID_30) && !defined(HAVE_SOLID_35)
+    FN("odbc_procedures", MAY_BE_RESOURCE|MAY_BE_FALSE),
+#endif
+#if !defined(HAVE_SOLID) && !defined(HAVE_SOLID_30) && !defined(HAVE_SOLID_35)
+    FN("odbc_foreignkeys", MAY_BE_RESOURCE|MAY_BE_FALSE),
+#endif
+    FN("odbc_specialcolumns", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("odbc_statistics", MAY_BE_RESOURCE|MAY_BE_FALSE),
+#if !defined(HAVE_DBMAKER) && !defined(HAVE_SOLID) && !defined(HAVE_SOLID_30) &&!defined(HAVE_SOLID_35)
+    FN("odbc_tableprivileges", MAY_BE_RESOURCE|MAY_BE_FALSE),
+#endif
+#if !defined(HAVE_DBMAKER) && !defined(HAVE_SOLID) && !defined(HAVE_SOLID_30) &&!defined(HAVE_SOLID_35)
+    FN("odbc_columnprivileges", MAY_BE_RESOURCE|MAY_BE_FALSE),
+#endif
     F1("pg_socket", MAY_BE_RESOURCE|MAY_BE_FALSE),
+    FN("pcntl_signal_get_handler", MAY_BE_CALLABLE|MAY_BE_LONG),
```

Comment:
MAY_BE_CALLABLE (and MAY_BE_ITERABLE) are not valid for func info.

Reply:
Thanks! Moved them to the `toOptimizerTypeMask()` method

Reply:
That's the wrong way around :) They *are* valid in toTypeMask(), but not in toOptimizerTypeMask() :) There MAY_BE_ITERABLE would be MAY_BE_ARRAY|MAY_BE_OBJECT and MAY_BE_CALLABLE would be MAY_BE_STRING|MAY_BE_ARRAY|MAY_BE_OBJECT|MAY_BE_ARRAY_KEY_LONG|MAY_BE_ARRAY_OF_STRING|MAY_BE_ARRAY_OF_OBJECT or so.

Reply:
Basically what you added as `@return string|array<int, string|object>|object` in some places. Possibly we'd want to replace those with `@return callable`, as that seems better for docs, even if it's not strictly true due to context-dependence?

Reply:
Ahh, I misunderstood your comment, and somehow thought that func info means arginfo.

Reply:
> Basically what you added as @return string|array<int, string|object>|object in some places. Possibly we'd want to replace those with @return callable, as that seems better for docs, even if it's not strictly true due to context-dependence?

Yeah, that's a good idea to do! And I can generate the right type mask for `callable`s.